http-service/src/main/java/net/runelite/http/service/hiscore/HiscoreService.java

```diff
@@ -67,53 +102,74 @@ private HiscoreResultBuilder lookupUsername(String username) throws IOException
 			.build();
 
 		Response okresponse = RuneliteAPI.CLIENT.newCall(okrequest).execute();
-		String responseStr;
 
-		try (ResponseBody body = okresponse.body())
+		if (!okresponse.isSuccessful())
 		{
-			responseStr = body.string();
+			switch (HttpStatus.valueOf(okresponse.code()))
+			{
+				case NOT_FOUND:
+					throw new NotFoundException();
+				default:
+					throw new RuntimeException("Error retrieving data from Jagex Hiscores: " + okresponse.message());
```

Comment:
Is throwing an unchecked exception the desired behavior here?

Reply:
I'm not sure how Spring handles these exceptions, would an IOException be better? I added this because I noticed that errors were getting logged in the Tomcat logs for 404's on accounts that weren't ironmen, etc

Reply:
You can define your own exceptions similar to https://github.com/runelite/runelite/blob/master/http-service/src/main/java/net/runelite/http/service/cache/NotFoundException.java